src/polytope/models/Problem.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass, field
+from typing import List
+
+from .ProblemStatement import ProblemStatement
+from .ProblemChecker import ProblemChecker
+from .ProblemValidator import ProblemValidator
+from .ProblemTest import ProblemPublicTest, ProblemTest
+from .ProblemSolution import ProblemSolution
+
+
+@dataclass
+class Problem:
+    """! A Polytope problem dataclass."""
+
+    """! Problem ID.
+
+    * It must be auto-generated by using UUID.
+    * It must be read-only.
+
+    TODO: Define UUID type and
+          set UUID generator as a default_factory.
+    """
+    _id: str
+
+    """! Problem name.
+
+    * It must be non-empty.
+    """
+    name: str
```

Comment:
(comment) length or character constraint on name?

Reply:
`name` is not used systematically, and it seems that it appears in `README.md` in the repository main folder.
There does not seem to be a need to restrict it right now, and I am not sure how to.

Reply:
I thought it appears in a directory name. Doesn't it need to be qualified as a valid dirname?

Reply:
I am planning on using `id` as the folder name instead of `name`.
I want `name` to be an unrestricted, human-readable string.

If we put a hyperlink to each problem folder in `README.md` of the repository main directory, I think it is okay even if the folder name is UUID.